File: Kalman_Filter_impl_v1/Filtre_Kalman_C++/kalman_test.py

# this code is for plotting the signal value with Noise + Signal after
# Using Kalman Filter 

import math
from random import random
import matplotlib.pyplot as plt
import numpy as np


input_file = open("noisy.txt", "r")
result_file = open("noisy_resultat.txt","r")

y1 = []
values_count = 0
for value in input_file:
    #print(value)
    y1.append(float(value.strip()))
    values_count+=1
print("Signal + Noise")

print(y1)

y2 = []
for value in result_file:
  #print(value)
  y2.append(float(value.strip()))

print("Signal after using Kalman Filter")
print(y2)

step = 1/(values_count/5)
x1 = np.arange(0, 5, step).tolist()
print(x1)

# plotting the points
plt.plot(x1, y1,label='signal_avec_noise')
plt.plot(x1, y2,label='filtered')
plt.title("Kalman Filter Example")
plt.ylabel("Temperature Sensor Reading (°C)")
plt.xlabel("Time (s)")
plt.legend()
plt.show()

File: Kalman_Filter_impl_v2/kalman_filter_matlab/kalman_test.m
clc;
clear all;
x = 1:1:34;

noisy_signal_file = fopen('noisy_temp_sensor_data.txt','r');
filtered_signal_file= fopen('filtered_temp_sensor_data.txt','w');

formatSpec = '%f\n';
sizeNOISY = Inf;
Message = "Noisy Temperature Sensor Data"
% read signal Data
[NOISY,count] = fscanf(noisy_signal_file,formatSpec,sizeNOISY)
FILTRED = zeros(count,1);
c = 1;
% initial states
% 	P : initial error covariance 
% 	K : initial Kalman gain
Ki = 0;
Pi = 10;
initial_estimate = 0;

for i = 1:count-1
    y = NOISY(i);
    % FILTRED : Estimated Value after Using Kalman Filter
    
    [FILTRED(c),Ki,Pi] = kalman_filter(NOISY(i),initial_estimate,Pi,[1],[1],[10],1);
    initial_estimate = FILTRED(c)
   
    % add FILTERED Value to Results File
    fprintf(filtered_signal_file,'%f\n',FILTRED(c));
    c = c+1;
end

% save Filtered Sensor Data into .txt File
fclose(noisy_signal_file);
fclose(filtered_signal_file);

% Plotting The Results
% X : Time (s)
x = [0:1/(count-1):1]
plot(x,FILTRED,x,NOISY);
title('kalman Filter Example')
legend('Filtered Signal ','Noisy Signal')

xlabel('Time (s)')
ylabel('Temperature Sensor Data (°C)')

